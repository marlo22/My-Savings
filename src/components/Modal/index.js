import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import { Modal as RNModal } from 'react-native';
import { Button, Icon } from 'native-base';

import { primaryColor, primaryContrastColor } from '../../themes';

const Header = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: ${primaryColor}
`;

const Title = styled.Text`
  font-size: 20px;
  font-weight: bold;
  padding: 0 10px;
  color: ${primaryContrastColor};
`;

const CloseButton = styled(Button)`
  padding: 0;
`;

const CloseIcon = styled(Icon)`
  color: ${primaryContrastColor};
`;

const Modal = ({ visible, title, onClose, children }) => (
  <RNModal visible={visible}>
    <Header>
      <Title>
        {title}
      </Title>
      <CloseButton onPress={onClose} transparent>
        <CloseIcon name="close" />
      </CloseButton>
    </Header>
    {children}
  </RNModal>
);

Modal.propTypes = {
  visible: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default Modal;