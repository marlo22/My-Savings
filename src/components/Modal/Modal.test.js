import React from 'react';
import { shallow } from 'enzyme';

import { Text } from 'react-native';
import Modal from './';

const modalDisplayName = 'Component';
const titleDisplayName = 'Styled(Text)';
const closeBtnDisplayName = 'Styled(Styled(Button))';

const visible = true;
const title = 'Modal title';
const onClose = jest.fn();
const children = <Text>Lorem ipsum</Text>;

const component = shallow(
  <Modal
    visible={visible}
    title={title}
    onClose={onClose}
  >
    {children}
  </Modal>
);

describe('<Modal />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should set Modal visible prop', () => {
    const modal = component.find(modalDisplayName);
    expect(modal.prop('visible')).toBe(visible);
  });

  test('should set passed title', () => {
    const titleComponent = component.find(titleDisplayName);
    expect(titleComponent.text()).toBe(title);
  });

  test('press close button should call onClose callback', () => {
    const closeBtn = component.find(closeBtnDisplayName);

    closeBtn.simulate('press');
    expect(onClose).toHaveBeenCalled();
  });

  test('should render passed children', () => {
    const childrenComponent = component.children().get(1);
    expect(childrenComponent).toEqual(children);
  });
});