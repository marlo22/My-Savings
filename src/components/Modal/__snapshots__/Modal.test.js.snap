// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> snapshot 1`] = `
<Component
  hardwareAccelerated={false}
  visible={true}
>
  <Styled(View)>
    <Styled(Text)>
      Modal title
    </Styled(Text)>
    <Styled(Styled(Button))
      onPress={[MockFunction]}
      transparent={true}
    >
      <Styled(Styled(Icon))
        name="close"
      />
    </Styled(Styled(Button))>
  </Styled(View)>
  <Text>
    Lorem ipsum
  </Text>
</Component>
`;
